src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
+            var segs = st.attrib("%s.segments".format(name));
+            if (segs.startsWith("Error:")) { 
+                var errorMsg = segs;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += "+created %s".format(segs);
+        }
+
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    /* 
+    Compile the component parts of a SegArray attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg name: name of SymTab element
+    :type name: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachSegArrayMsg(cmd: string, name: string, st: borrowed SymTab): MsgTuple throws {
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting SegArray components for '%s'".format(cmd, name));
+
+        var repMsg: string;
+
+        var segs = st.attrib("%s_segments".format(name));
+        if (segs.startsWith("Error:")) { 
+            var errorMsg = segs;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        var vals = st.attrib("%s_values".format(name));
+        if (vals.startsWith("Error:")) { 
+            var errorMsg = vals;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        var lens = st.attrib("%s_lengths".format(name));
+        if (lens.startsWith("Error:")) { 
+            var errorMsg = lens;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        repMsg = "segarray+created %s+created %s+created %s".format(segs, vals, lens);
+
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    /* 
+    Parse, execute, and respond to a generic attach message
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg payload: request containing (dtype+name)
+    :type payload: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc genAttachMsg(cmd: string, payload: string, 
+                                            st: borrowed SymTab): MsgTuple throws {
+        var repMsg: string; // response message
+
+        var ele_parts = payload.split("+");
+        var dtype = ele_parts[0];
+        var name = ele_parts[1];
+
+        if dtype == "infer" {
+            // Try to determine the type from the entries in the symbol table
+            // Should try __RegisteredSymbols__ instead
+            var symbols = st.info("__RegisteredSymbols__");
```

Comment:
do you ever actually use `symbols`

Reply:
No, that was leftover from debugging - good catch